mlflow/utils/rest_utils.py

```diff
@@ -77,6 +79,60 @@ def http_request(
     Returns:
         requests.Response object.
     """
+    hostname = host_creds.host
+    cleaned_hostname = strip_suffix(hostname, "/")
+    url = f"{cleaned_hostname}{endpoint}"
+
+    if host_creds.auth_by_databricks_sdk:
+        from databricks.sdk import WorkspaceClient
+        from databricks.sdk.config import Config
+        from databricks.sdk.errors import DatabricksError
+
+        config = Config(
+            profile=host_creds.databricks_auth_profile,
+            retry_timeout_seconds=MLFLOW_DATABRICKS_ENDPOINT_HTTP_RETRY_TIMEOUT.get(),
+        )
+        # Note: If we use `config` param, all SDK configurations must be set in `config` object.
+        ws_client = WorkspaceClient(config=config)
+        try:
+            # Databricks SDK `APIClient.do` API is for making request using
+            # HTTP
+            # https://github.com/databricks/databricks-sdk-py/blob/a714146d9c155dd1e3567475be78623f72028ee0/databricks/sdk/core.py#L134
+            # The 'query' argument of Databricks SDK `APIClient.do` API is equivalent to
+            # `params` argument of `requests.Session.request()` API
+            # The 'body' argument of Databricks SDK `APIClient.do` API is equivalent to
+            # `json` argument of `requests.Session.request()` API
+            raw_response = ws_client.api_client.do(
+                method=method,
+                path=endpoint,
+                headers=extra_headers,
+                raw=True,
+                query=kwargs.get("params"),
+                body=kwargs.get("json"),
+                files=kwargs.get("files"),
+                data=kwargs.get("data"),
+            )
+            return raw_response["contents"]._response
+        except DatabricksError as e:
+            if hasattr(e, "error_code"):
```

Comment:
Does `DatabricksError` always have `error_code` or not?

Reply:
Yes. `DatabricksError` class constructor always set `self. error_code`